sui/src/sui_commands.rs

```diff
@@ -29,53 +30,112 @@ use crate::config::{
-        config: PathBuf,
+        #[structopt(long)]
+        config: Option<PathBuf>,
     },
     #[structopt(name = "genesis")]
     Genesis {
-        #[structopt(long, default_value = ".")]
-        working_dir: PathBuf,
         #[structopt(long)]
-        config: Option<PathBuf>,
+        working_dir: Option<PathBuf>,
+        #[structopt(short, long, help = "Forces overwriting existing configuration")]
+        force: bool,
     },
 }
 
+pub fn sui_config_dir() -> Result<PathBuf, anyhow::Error> {
+    let mut p = match dirs::home_dir() {
+        Some(v) => v,
+        None => return Err(anyhow!("Cannot obtain home directory path")),
+    };
+    p.push(SUI_DIR);
+    p.push(SUI_CONFIG_DIR);
+    Ok(p)
+}
+
 impl SuiCommand {
     pub async fn execute(&self) -> Result<(), anyhow::Error> {
         match self {
             SuiCommand::Start { config } => {
-                let config: NetworkConfig = PersistedConfig::read(config)?;
+                let config_path = match config {
+                    Some(v) => v.clone(),
+                    None => {
+                        let mut p = sui_config_dir()?;
+                        p.push(SUI_NETWORK_CONFIG);
+                        p
+                    }
+                };
+                let config: NetworkConfig = PersistedConfig::read(&config_path).map_err(|err| {
+                    err.context(format!(
+                        "Cannot open Sui network config file at {}",
+                        config_path.to_str().unwrap()
+                    ))
+                })?;
                 SuiNetwork::start(&config)
                     .await?
                     .wait_for_completion()
                     .await
             }
-            SuiCommand::Genesis {
-                working_dir,
-                config: path,
-            } => {
-                let network_path = working_dir.join("network.conf");
-                let wallet_path = working_dir.join("wallet.conf");
-                let keystore_path = working_dir.join("wallet.key");
-                let db_folder_path = working_dir.join("client_db");
-
-                if let Ok(config) = PersistedConfig::<NetworkConfig>::read(&network_path) {
-                    if !config.authorities.is_empty() {
-                        return Err(anyhow!("Cannot run genesis on a existing network, please delete network config file and try again."));
+            SuiCommand::Genesis { working_dir, force } => {
+                let sui_config_dir = &match working_dir {
+                    Some(v) => v.clone(),
+                    None => {
+                        // create default Sui config dir if not specified
+                        // on the command line and if it does not exist
+                        // yet
+                        let config_path = sui_config_dir()?;
+                        fs::create_dir_all(&config_path)?;
+                        config_path
```

Comment:
We can condense this down if we create the dir after handled the Optional value, which we probably wants to do anyway to prevent failure when user provide a non-existent path?

```suggestion
                let sui_config_dir = working_dir.clone().unwrap_or(sui_config_dir()?);
                fs::create_dir_all(&sui_config_dir)?;
```

Reply:
Actually, this logic was intentional. I am not sure we should create a directories in case a path is specified by the user, but rather fail if it does not exist. This feels more natural to me, so I am going to leave this as is, but I am not religious about it either.